tests/integrationv2/test_signature_algorithms.py

```diff
@@ -25,9 +24,31 @@
     Signatures.ECDSA_SECP256r1_SHA256,
     Signatures.RSA_PSS_RSAE_SHA256,
     Signatures.RSA_PSS_PSS_SHA256,
+    Signatures.ECDSA_SHA224,
+    Signatures.ECDSA_SHA1,
 ]
 
 
+def expected_signature(protocol, signature):
+    if protocol < Protocols.TLS12:
+        if signature.sig_type == 'ECDSA':
+            signature = Signatures.ECDSA_SHA1
+        else:
+            signature = Signatures.RSA_MD5_SHA1
+    return signature
+
+
+def expected_signature_alg_tls12(signature):
+    # ECDSA by default hashes with SHA-1.
+    #
+    # This is inferred from the rfc- https://www.rfc-editor.org/rfc/rfc4492#section-5.10
```

Comment:
Isn't this RFC an extension to TLS 1.1? Was this meant for the other function?

Also, this function seems very specific to the new test that you added, since it's checking for RSA_SHA224 specifically in order to determine what to fallback to. Since this function isn't being used anywhere else, I think this part might be easier to understand if it was just in the test itself rather than a function.